### ISSUE TITLE
SSH Public key authentication fails in docker env

### ISSUE THREAD

**Issue by phoenix0984** [ytti/oxidized#1765]
When trying to authenticate via privatekey, authentication fails with:

`D, [2019-03-30T12:27:07.214263 #228] DEBUG -- net.ssh.authentication.session[2ae5fc0fb010]: trying publickey
D, [2019-03-30T12:27:07.214699 #228] DEBUG -- net.ssh.authentication.agent[2ae5fc0f7b40]: connecting to ssh-agent
E, [2019-03-30T12:27:07.214885 #228] ERROR -- net.ssh.authentication.agent[2ae5fc0f7b40]: could not connect to ssh-agent: Agent not configured
`

However logging in manually from inside the container works perfectly fine:

`mg@docker-swarm-n2:~$ docker exec -it oxidized_oxidized.1.r3kw4as9wt0eo7ej0qvld76s0 /bin/bash -c "ssh rancid@10.2.3.2 -v -i /root/.config/oxidized/.ssh/id_rsa"
`

This happens on oxidized/oxidized:latest. If any more input is needed, I´m happy to help out, if I can.

Kind regards

**Comment by RiwanBodereau** [comment]
Hello,

It should not use the ssh-agent, it means net::ssh couldn't find your private key.

I had the same problem, please make sure you specify your private key correctly, in `ssh_keys` variable :

```
vars:
  auth_methods: [ "none", "publickey", "password"]
  ssh_keys: "/root/.config/oxidized/.ssh/id_rsa"
```

Regards,

**Comment by phoenix0984** [comment]
@RiwanBodereau 
Well, thanks for your suggestion. `ssh_keys` did the job!
All I got when googling for key-based authentication was the variable `privatekey` and that obviously didn´t play well ;)

This definetly needs to be in the docs. Let´s see, if I could make a PR for that...

**Comment by laf** [comment]
A pull request would be great if you can submit one please :)

**Comment by phoenix0984** [comment]
@laf 
Pull request has been created. It´s just quick & dirty, so let me know if this needs some rework.

**Comment by wk** [comment]
Closed via #1767. Thanks @phoenix0984 and @laf!

**Comment by laf** [comment]
I merely directed :)
